CMakeLists.txt

```diff
@@ -565,6 +583,13 @@ set(SOURCES
         utilities/write_batch_with_index/write_batch_with_index_internal.cc
         $<TARGET_OBJECTS:build_version>)
 
+if(HAVE_POWER8)
+  list(APPEND SOURCES
```

Comment:
also, i think the ppc detection logic won't work if `HAVE_POWER8` and `HAS_ALTIVEC` are not defined. in `Makefile`, they are added to `CXXFLAGS` and `CFLAGS` if the C++ complier supports them.

we might want to do the same on the cmake side, like

```cmake
  set_source_files_properties(
    util/crc32c.cc
    util/crc32c_ppc.cc
    PROPERTIES COMPILE_FLAGS -DHAVE_POWER8)
```

and may i learn merit of defining `POWER8` w/o defining `HAS_ALTIVEC`? if the only purpose is to print the arch in logging message, is it a little bit overkill to do so?

Reply:
Yes, we should append in a single shot and add the PROPERTIES, as you mentioned.  

Why do we define POWER8 and HAS_ALTIVEC separately?  The crc optimized assembly uses altivec instructions that were present on Power7 and earlier hardware levels.  The Power8 hardware level introduced special parallel vector processing assembly instructions which are also used in the crc optimized assembly.  In general, we want to be clear about calling out these two different dependencies because we have so many different combinations to consider that things can easily get a bit confusing (Power7 or earlier case, Powe8 or later case, big endian case, little endian case).  Most specifically, we need to be clear about the dependencies in case (a) someone wants to continue to build / run on Power7 or earlier hardware or (b) someone wants to introduce Power architecture code to Ceph that has nothing to do with crc, or Power8 parallel vector processing, or altivec.

Reply:
the commend regarding to `-DHAVE_POWER8` is not addressed.